### DISCUSSION TITLE
So is SNES supported?

### DISCUSSION THREAD

**Discussion by MrVeink** [cc65/cc65#1886] [Q&A]
Was trying to find an answer for this and I did find that ca65 can assemble for SNES but can you do C and have SNES as the target? If not, is anyone into that?

From [georgjz's guide](https://georgjz.github.io/snesaa01/):
```
MEMORY
{
    ROM0: start = $008000, size = $8000, fill = yes;
}

SEGMENTS
{
    CODE:   load = ROM0, align = $100;
    VECTOR: load = ROM0, start = $00ffe4;
}
```
Was looking at the [config files](https://github.com/cc65/cc65/tree/master/cfg) in the source and didn't find an SNES file so this above could help.

**Comment by polluks** [comment]
10 lines for a new target? Let's take a look
https://cc65.github.io/doc/customizing.html
FYI https://forums.nesdev.org/viewtopic.php?t=18361
Alas, cc65 is not a 65816 compiler right now.

**Comment by Fabrizio-Caruso** [comment]
@polluks if the 65816 can run plain 65(c)02 as I suspect, then cc65 could be used to compile sub-optimal code for the SNES. It would be better than nothing but not better than the already available C compilers that target the 65816.

**Reply by SeanRamey** [reply]
Which C compilers are available that target 65816? I'm finding an incredibly out of date and sparse list. Also, it seems that this guy started working on supporting the 65816 in C: https://github.com/bauen1/cc65

**Reply by polluks** [reply]
A Tiny C Compiler for 65816 CPU from the SNES-SDK https://github.com/nArnoSNES/tcc-65816.

**Reply by SeanRamey** [reply]
Is this the SDK you are referring to? https://github.com/Shad0w1nk/snes-sdk
https://code.google.com/archive/p/snes-sdk/
https://github.com/alesegdia/snes-sdk-linux
https://github.com/optixx/snes-sdk

If so, is that tcc-65816 github just an archive of the original version from the SDK or what? I had found that originally, but I was under the impression that since it was only the "Initial commit" and it was archived, that it wasn't actually working.

**Reply by SeanRamey** [reply]
Also, just found out that none of these versions seem to be actually building at the moment. I've tried them and they fail the build, or in the case of tcc-65816, it says there's nothing to be done.
